File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int result = 0;
        // HashMap<Integer,Integer> map = new HashMap<>();
        // for(int i:nums)
        // {
        //     if(map.get(i)==null)
        //     {
        //         map.put(i,map.getOrDefault(i-1,0)+map.getOrDefault(i+1,0)+1);
        //         map.put(i-map.getOrDefault(i-1,0),map.get(i));
        //         map.put(i+map.getOrDefault(i+1,0),map.get(i));
        //         result = Math.max(result,map.get(i));
        //     }

        // }

        HashSet<Integer> hs = new HashSet<>();
        for(int i:nums)
        {
            hs.add(i);
        }
        for(int i:hs)
        {
            if(!hs.contains(i-1))
            {
                int count = 1;
                int num = i;
                while(hs.contains(num+1))
                {
                    count++;
                    num++;
                }
                result = Math.max(count,result);
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int result=0;
        int highest = prices[prices.size()-1];
        for(int i=prices.size()-2;i>=0;i--)
        {
            if(prices[i]>highest)
            {
                highest = prices[i];
            }
            else
            {
                result = max(result,highest-prices[i]);
            }
        }
        return result;
    }
};
